packages/algob/src/lib/algo-operator.ts

```diff
@@ -54,18 +62,103 @@ export class AlgoOperatorImpl implements AlgoOperator {
     }
   };
 
+  getTxFee (params: algosdk.SuggestedParams, txSize: number): number {
+    if (params.flatFee) {
+      return Math.max(ALGORAND_MIN_TX_FEE, params.fee);
+    }
+    return Math.max(ALGORAND_MIN_TX_FEE, txSize);
+  }
+
+  getUsableAccBalance (accoutInfo: algosdk.AccountInfo): number {
+    // Extracted from interacting with Algorand node:
+    // 7 opted-in assets require to have 800000 micro algos (frozen in account).
+    // 11 assets require 1200000.
+    return accoutInfo.amount - (accoutInfo.assets.length + 1) * ALGORAND_ASA_OWNERSHIP_COST;
+  }
+
+  getOptInTxSize (params: algosdk.SuggestedParams, accounts: Accounts): number {
+    const randomAccount = accounts.values().next().value;
+    // assetID can't be known before ASA creation
+    // it shouldn't be easy to find out the latest asset ID
+    // In original source code it's uint64:
+    // https://github.com/algorand/go-algorand/blob/1424855ad2b5f6755ff3feba7e419ee06f2493da/data/basics/userBalance.go#L278
+    const assetID = Number.MAX_SAFE_INTEGER; // not 64 bits but 55 bits should be enough
+    const sampleASAOptInTX = tx.makeASAOptInTx(randomAccount.addr, assetID, params);
+    const rawSignedTxn = sampleASAOptInTX.signTxn(randomAccount.sk);
+    return rawSignedTxn.length;
+  }
+
+  async optInToASA (
+    asaName: string, assetIndex: number, account: Account, params: algosdk.SuggestedParams
+  ): Promise<void> {
+    console.log(`ASA ${account.name} opt-in for for ASA ${asaName}`);
+    const sampleASAOptInTX = tx.makeASAOptInTx(account.addr, assetIndex, params);
+    const rawSignedTxn = sampleASAOptInTX.signTxn(account.sk);
+    const txInfo = await this.algodClient.sendRawTransaction(rawSignedTxn).do();
+    await this.waitForConfirmation(txInfo.txId);
+  }
+
+  async optInToASAMultiple (
+    asaName: string, assetIndex: number, accounts: Account[], params: algosdk.SuggestedParams
+  ): Promise<void> {
+    for (const account of accounts) {
+      await this.optInToASA(asaName, assetIndex, account, params);
+    }
+  }
+
+  async checkASAOptInAccountBalances (
```

Comment:
let's rename to `checkBalanceForOptInTx`

Reply:
OK, but we don't need `ASA` infix (there no other thing we can opt-it).

Reply:
So we could have it a bit shorter ;)

Reply:
`checkBalanceForOptInTx`